scripts/validate-voice-channel-names.js

```diff
@@ -0,0 +1,63 @@
+import { readFileSync } from 'fs';
+import * as path from 'path';
+import * as url from 'url';
+
+/** Current directory */
+const CURRENT_DIR = url.fileURLToPath(new URL('.', import.meta.url));
+
+/** Relative file path to parse. */
+const FILE_PATH = '../lists/voice-channel-names.txt';
+
+/**
+ * List of filter conditions to vlidate when parsing the file. If the result
+ * is false, it will remove the line.
+ *
+ * @type {readonly ((str: string) => boolean)[]}
+ */
+const FILTERS = [(str) => !str.startsWith('//'), (str) => str.length];
+
+/**
+ * List of validation functions to run against all lines in the file. If the
+ * result is false, it will throw an error.
+ *
+ * @type {readonly ((str: string) => boolean)[]}
+ */
+const VALIDATORS = [(str) => str.length < 30];
+
+// Parse & filter groups from file
+const groups = readFileSync(path.join(CURRENT_DIR, FILE_PATH), {
+  encoding: 'utf-8'
+})
+  .split(/[\n]{2,}/gm)
+  .map((group) => group.split('\n'))
+  .map((group) =>
+    group.filter((name) => FILTERS.every((filter) => filter(name)))
+  )
+  .filter((group) => group.length !== 0);
+
+/**
+ * Gets list of errors based on validators.
+ *
+ * @type {string[]}
+ */
+const errors = [];
+for (const group of groups) {
+  for (const name of group) {
+    const valid = VALIDATORS.every((validator) => validator(name));
+
+    if (!valid) {
+      errors.push(
+        `ERROR: ${name} is not in a valid format. All names must be less then 30 characters`
```

Comment:
Not the biggest fan of this being an error. [Discord supports up to 100 characters in channel names](https://discord.com/developers/docs/resources/channel), and there are opportunities for humor in a long name that is cut off.

Potential solution: Comment at the top of the file saying that names longer than 30 characters might not be accepted, and it's enforced at the PR level. Not ideal, but workable (especially since we're gonna want to vet channel names anyways for TOS-related reasons

Reply:
Fair enough. I'll raise it to 100 for the automated check & add a comment at the top of the text file.